cdap-ui/app/features/hydrator/services/create/stores/config-store.js

```diff
@@ -545,6 +545,20 @@ class ConfigStore {
         });
       });
     }
+    errorFactory.allNodesConnected(nodes, connections, (err, nodes) => {
+      if (err) {
+        isStateValid = false;
+
+        angular.forEach(nodes, (node) => {
```

Comment:
This might be moving away from the pattern.

The pattern that we used to follow was, if its a dag level validation then we call the callback with the error.

If its a node level validation the validation function calls the callback for each node. If there is an error we either choose to show or hide the error and modify the model appropriately. 

So instead of forming the ```errors``` array in the callback we need to call the callback with error for each node. This would avoid two ```angular.forEach``` (in validation function as well as here).